middlewares/valueVerifier.js

```diff
@@ -47,8 +78,12 @@ class valueVerifier {
    */
   static async verifyReactionId(req, res, next) {
     try {
-      const reactionId = await findReaction(req.params.reactionId);
-      if (reactionId) {
+      if (req.baseUrl === COMMENT_BASE_URL) {
+        reaction = await findReaction(req.params.reactionId);
+      } else if (req.baseUrl === ARTICLE_BASE_URL) {
+        reaction = await getReaction(req.params.reactionId);
+      }
+      if (reaction) {
```

Comment:
Please, can you explain what is going on here?

Reply:
I finally made use of closure and it's cleaner. Thanks for the help